test_rclcpp/test/test_subscription.cpp

```diff
@@ -140,3 +140,50 @@ TEST(CLASSNAME(test_subscription, RMW_IMPLEMENTATION), subscription_and_spinning
   executor.spin_node_some(node);
   ASSERT_EQ(5, counter);
 }
+
+// Shortened version of the test for the ConstSharedPtr callback signature
+TEST(CLASSNAME(test_subscription, RMW_IMPLEMENTATION), subscription_shared_ptr_const) {
+  rclcpp::init(0, nullptr);
+
+  auto node = rclcpp::Node::make_shared("test_subscription");
+
+  rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_default;
+  custom_qos_profile.depth = 10;
```

Comment:
That seems to be the default now anyway? https://github.com/ros2/rmw/blob/65a952913a49b0f9c38088a8dc0504b6943ed609/rmw/include/rmw/qos_profiles.h#L39-L42